src/main/java/com/team4/testingsystem/services/impl/QuestionServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.team4.testingsystem.services.impl;
+
+import com.team4.testingsystem.entities.Level;
+import com.team4.testingsystem.entities.Module;
+import com.team4.testingsystem.entities.Question;
+import com.team4.testingsystem.entities.User;
+import com.team4.testingsystem.exceptions.NotFoundException;
+import com.team4.testingsystem.repositories.QuestionRepository;
+import com.team4.testingsystem.services.QuestionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class QuestionServiceImpl implements QuestionService {
+
+    private final QuestionRepository questionRepository;
+    private final UsersServiceImpl usersService;
+    private final LevelServiceImpl levelService;
+    private final ModuleServiceImpl moduleService;
+
+    @Autowired
+    QuestionServiceImpl(QuestionRepository questionRepository, UsersServiceImpl usersService, LevelServiceImpl levelService, ModuleServiceImpl moduleService) {
+        this.questionRepository = questionRepository;
+        this.usersService = usersService;
+        this.levelService = levelService;
+        this.moduleService = moduleService;
+    }
+
+    @Override
+    public Question getQuestionById(Long id) {
+        Question question = questionRepository.findById(id).orElse(null);
+        if (question == null) throw new NotFoundException();
+        return question;
+    }
+
+    @Override
+    public Question createQuestion(String questionBody, boolean isAvailable, Long creatorUserId, Long levelID, Long moduleID) {
+        Question question = new Question();
+        question.setQuestionBody(questionBody);
+        question.setAvailable(isAvailable);
+        User user = usersService.getUserById(creatorUserId);
+        question.setCreator(user);
+        Level level = levelService.getLevelById(levelID);
+        question.setLevel(level);
+        Module module = moduleService.getModuleById(moduleID);
+        question.setModule(module);
+        questionRepository.save(question);
```

Comment:
I really don't like an idea of such a long initialization. Especially as it is divided by getting other objects. Can we create a constructor and make it in one line?

Reply:
I was told that a constructor with 3 or more parameters it is bad practice, can I use lombok for this?

Reply:
Any method with a big amount of parameters is bad practice.
The reason here is that you may have a mistake in the ordering of your parameters, if they have the same type.
Lombok will generate exactly the same constructor. So it is not a good fix.

At the same time, splitting the initialization is a really bad practice as well. It can be really hard to find all the places where you are using it, for example, if you need to add a new parameter.

From that perspective, I would say that it's better to violate the first rule, than the second.
Alternatively, you can create a builder for the object. I think it's one of the best fixes to the problem.